server/src/main/java/keywhiz/service/providers/ClientAuthenticator.java

```diff
@@ -160,28 +160,41 @@ public ClientAuthenticator(
 
     // The sub-lists returned by getSubjectAlternativeNames have an integer for the first
     // entry, representing a field, and the value as a string as the second entry.
-    List<String> spiffeUriNames = sans.stream()
+    List<String> providedUris = sans.stream()
         .filter(sanPair -> sanPair.get(0).equals(URINAME_SAN))
         .map(sanPair -> (String) sanPair.get(1))
+        .collect(Collectors.toUnmodifiableList());
+
+    // https://spiffe.io/spiffe/concepts/#spiffe-verifiable-identity-document-svid
+    // > An SVID contains a single SPIFFE ID, which represents the identity of the service presenting it
+    //
+    // https://github.com/spiffe/spiffe/blob/master/standards/X509-SVID.md#2-spiffe-id
+    // > An X.509 SVID MUST contain exactly one URI SAN.
+    List<String> spiffeUriNames = providedUris.stream()
         .filter(uri -> uri.startsWith(SPIFFE_SCHEME))
         .collect(Collectors.toUnmodifiableList());
 
     if (spiffeUriNames.size() > 1) {
       logger.warn("Got multiple SPIFFE URIs from certificate: {}", spiffeUriNames);
       return Optional.empty();
+    } else if (spiffeUriNames.size() == 1 && providedUris.size() > 1) {
```

Comment:
It would be good to have a test for a cert that has one SPIFFE ID and multiple URI SANs